src/NUnitTestAdapter/TestConverterForXml.cs

```diff
@@ -0,0 +1,415 @@
+// ***********************************************************************
+// Copyright (c) 2011-2020 Charlie Poole, Terje Sandstrom
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
+using NUnit.VisualStudio.TestAdapter.NUnitEngine;
+using VSTestResult = Microsoft.VisualStudio.TestPlatform.ObjectModel.TestResult;
+
+namespace NUnit.VisualStudio.TestAdapter
+{
+    public interface ITestConverterCommon
+    {
+        TestCase GetCachedTestCase(string id);
+        TestConverterForXml.TestResultSet GetVsTestResults(INUnitTestEventTestCase resultNode, ICollection<INUnitTestEventTestOutput> outputNodes);
+    }
+
+    public interface ITestConverter : ITestConverterCommon
+    {
+        /// <summary>
+        /// Converts an NUnit test into a TestCase for Visual Studio,
+        /// using the best method available according to the exact
+        /// type passed and caching results for efficiency.
+        /// </summary>
+        TestCase ConvertTestCase(NUnitDiscoveryTestCase testNode);
+    }
+
+    public interface ITestConverterXml : ITestConverterCommon
+    {
+        TestCase ConvertTestCase(NUnitEventTestCase nUnitEventTestCase);
+    }
+
+    public sealed class TestConverterForXml : IDisposable, ITestConverterXml
```

Comment:
It seems like there's a lot in common with the TestConverter.cs file and I'm worried the duplication will be a future maintenance hazard.

Reply:
I am planning of leaving this class behind, as legacy, and not move it forward.  The only way to get to it, is to set the DiscoveryMethod as Legacy.  As the new code proves that it works, then this code should be deprecated and then removed.

Reply:
Sounds good.